internal/pkg/cli/pipeline_init.go

```diff
@@ -355,27 +353,56 @@ func relPath(fullPath string) string {
 	return relPath
 }
 
-// TODO: Nice-to-have: have an opts.listRemoteRepos() method that execs out to `git remote -v` and parse repo name to offer select menu
-func (opts *InitPipelineOpts) selectGitHubRepo() error {
-	repo, err := opts.prompt.Get(
-		pipelineEnterGitHubRepoPrompt,
-		fmt.Sprintf(`The GitHub repository linked to your workspace. Pushing to this repository will trigger your pipeline build stage. Please enter full repository URL, e.g. "https://github.com/myCompany/myRepo", or the owner/rep, e.g. "myCompany/myRepo"`),
-		validateGitHubRepo,
+func (opts *InitPipelineOpts) selectGitHubURL() error {
+	url, err := opts.prompt.SelectOne(
+		pipelineSelectGitHubURLPrompt,
+		pipelineSelectGitHubURLHelpPrompt,
+		opts.repoURLs,
 	)
-
-	// Remove ".git" at the end of the Github repo URL.
-	repoNameSplit := strings.Split(repo, ".")
-	if repoNameSplit[len(repoNameSplit)-1] == "git" {
-		repo = strings.Join(repoNameSplit[:len(repoNameSplit)-1], ".")
+	if err != nil {
+		return fmt.Errorf("select GitHub URL: %w", err)
 	}
+	opts.GitHubURL = url
 
-	if err != nil {
-		return fmt.Errorf("failed to get GitHub repository: %w", err)
+	return nil
+}
+
+func (opts *InitPipelineOpts) parseOwnerRepoName(url string) (string, string, error) {
+	parsedURL := strings.TrimPrefix(url, opts.regexZeroLengthAssertion(url, `.*(github.com)(:|\/)`, "", ""))
+	ownerRepo := strings.Split(parsedURL, string(os.PathSeparator))
+	if len(ownerRepo) != 2 {
+		return "", "", fmt.Errorf(`unable to parse the owner and repository for "%s", please pass in your repository URL with -u`, url)
```

Comment:
If they pass the url with the flag, it will crash again the same way no? Assuming they'll just re-enter the same url.

Maybe, instead we should give them an example of an accepted format? 
```
unable to parse the GitHub repository owner and name from %s: please pass the repository URL with the format `--url https://github.com/{owner}/{repositoryName}`
```

Reply:
Not necessarily. If their git remote result like this:

```
efekarakus whatever git@github.com:efekarakus/grit.git (fetch)
```

this one fails. But if they pass the url `git@github.com:efekarakus/grit.git` using flag, it will not fail. However, if the url itself is invalid then it should fail again. I agree we can add an example :+1: